### ISSUE TITLE
schema cache does not work properly with postgres virtual columns

### ISSUE THREAD

**Issue by netrusov** [rails/rails#46935]
sample project: https://github.com/netrusov/rails-schema-cache-issue/tree/master

### Steps to reproduce

create project from scratch
```shell script
rails new dummy -d postgresql -M -C -A -J -T --api
```

add migration
```ruby
class CreateUsers < ActiveRecord::Migration[7.0]
  def change
    create_table :users do |t|
      t.text :first_name
      t.text :last_name
      t.virtual :full_name, type: :text, as: "first_name || ' ' || last_name", stored: true

      t.timestamps
    end
  end
end
```

after migration the `db/schema.rb` should contain valid schema
```ruby
ActiveRecord::Schema[7.0].define(version: 2023_01_09_121035) do
  create_table "users", force: :cascade do |t|
    t.text "first_name"
    t.text "last_name"
    t.virtual "full_name", type: :text, as: "((first_name || ' '::text) || last_name)", stored: true
    t.datetime "created_at", null: false
    t.datetime "updated_at", null: false
  end
end
```

dump schema cache and then dump schema again (note that this must be done in separate commands)
```shell script
bin/rails db:schema:cache:dump
bin/rails db:schema:dump
```

### Expected behavior

`db/schema.rb` must remain the same

### Actual behavior

`db/schema.rb` contains wrong info about virtual columns
```ruby
ActiveRecord::Schema[7.0].define(version: 2023_01_09_121035) do
  create_table "users", force: :cascade do |t|
    t.text "first_name"
    t.text "last_name"
    t.text "full_name", default: -> { "((first_name || ' '::text) || last_name)" } # <--- HERE
    t.datetime "created_at", null: false
    t.datetime "updated_at", null: false
  end
end
```

and `PG::FeatureNotSupported` error occurs while trying to apply schema
```
$ bin/rails db:reset
Ignoring db/schema_cache.yml because it has expired. The current schema version is 0, but the one in the schema cache file is 20230109121035.
Dropped database 'dummy_development'
Dropped database 'dummy_test'
Created database 'dummy_development'
Created database 'dummy_test'
rails aborted!
ActiveRecord::StatementInvalid: PG::FeatureNotSupported: ERROR:  cannot use column reference in DEFAULT expression
LINE 1: ...ext, "last_name" text, "full_name" text DEFAULT ((first_name...
```

### System configuration
**Rails version**: 7.0.4

**Ruby version**: 3.2.0

**Comment by fatkodima** [comment]
Thanks for reporting. I will look into that.

**Comment by estebanz01** [comment]
@fatkodima hey! sorry to comment on a closed, fixed issue, but do you happen to know when the fix for this issue is going to be released? I don't see this in v7.0.4.3

**Comment by byroot** [comment]
I forgot to backport it. I just did it as 302aa41a07956d58169376bdd4543d2e09811d48

You can point your Gemfile at the `7-0-stable` branch until the next release happens.
